tripper/datadoc/keywords.py

```diff
@@ -178,100 +251,680 @@ def add_theme(
+            resources[iriname(k)] = d
+
+        # Add properties
+        for propname, value in properties.items():
+            name = iriname(propname)
+            label = value["label"] if "label" in value else name
+            d = AttrDict(iri=value["@id"])
+            if "@type" in value:
+                d.type = to_prefixed(value["@type"], p)
+            d.domain = value.get("domain", RDFS.Resource)
+
+            for domain in asseq(d.domain):
+                domainname = iriname(domain)
+                if domainname not in resources:
+                    if domainname not in self.data.resources:
+                        if domainname not in ("Resource",):
+                            warnings.warn(
+                                f"Adding undefined domain '{domain}' for "
+                                f"keyword '{label}'",
+                                MissingKeywordsClassWarning,
+                            )
+                        r = AttrDict(
+                            iri=prefix_iri(domain, self.data.prefixes),
+                            keywords=AttrDict(),
+                        )
+                    else:
+                        r = self.data.resources[domainname].copy()
+                    resources[domainname] = r
+                    r.keywords[label] = d
+                else:
+                    resources[domainname].keywords[label] = d
+            if "range" in value:
+                _types = asseq(d.get("type", OWL.AnnotationProperty))
+                types = [expand_iri(t, p) for t in _types]
+                if OWL.ObjectProperty in types:
+                    d.range = value["range"]
+                else:
+                    d.range = "rdfs:Literal"
+                    if "range" in value:
+                        d.datatype = value["range"]
+            else:
+                d.range = "rdfs:Literal"
+                # TODO: Define if we accept missing datatype for literals
+            if "conformance" in value:
+                d.conformance = CONFORMANCE_MAPS[value["conformance"]]
+            if "unitSymbol" in value:
+                d.unit = value["unitSymbol"]
+            for k, v in value.items():
+                if (
+                    k not in ("@id", "@type", "domain", "label", "name")
+                    and k not in d
+                ):
+                    d[k] = v
+
+        return data
+
+    def missing_keywords(
+        self,
+        ts: "Triplestore",
+        include_classes: bool = False,
+        return_existing: bool = False,
+    ) -> "Union[list, Tuple[list, list]]":
+        """List keywords not defined in triplestore `ts`.
+
+        Arguments:
+            ts: Triplestore object to check.
+            include_classes: Also return missing classes.
+            return_existing: If true, two lists are returned:
+                - list of keywords missing `ts`
+                - list of keywords existing `ts`
```

Comment:
Does not make sensee

Reply:
I see there is a typo. Apart from that, what doesn't make sense?